### PR TITLE
Feature #127 image content

### LINKED ISSUE #127: Bot should be able to answer with images.
As a bot management user, I would like to be able to make bot answer to some intents with images.

### PR DESCRIPTION
**Pull request by platonovsimeon**
resolves #127

### CODE REVIEW

File: src/main/java/fi/metatavu/metamind/images/ImageServlet.java [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+  private void setCorsHeaders(HttpServletResponse resp) {
+    resp.setHeader("Access-Control-Allow-Origin", "*");
+    resp.setHeader("Access-Control-Allow-Headers", "Origin, X-Requested-With, Content-Type, Accept");
+    resp.setHeader("Access-Control-Allow-Methods", "GET,PUT,POST,DELETE,OPTIONS");
+  }
+  @Override 
+  protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws IOException{
+    setCorsHeaders(resp);
+    String fileURL = req.getPathInfo();
+    File file = new File("images/"+fileURL);
```
Comment (hkurhinen) [inline src/main/java/fi/metatavu/metamind/images/ImageServlet.java]: Make folder configurable with for example system property

File: src/main/java/fi/metatavu/metamind/images/ImageServlet.java [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+    } catch (IOException e) {
+      logger.warn("Failed to send response", e);
+    }
+  }
+  
+  @Override 
+  protected void doDelete(HttpServletRequest req, HttpServletResponse resp) throws IOException{
+    setCorsHeaders(resp);
+    String fileURL = req.getPathInfo();
+    File file = new File("images/"+fileURL);
```
Comment (hkurhinen) [inline src/main/java/fi/metatavu/metamind/images/ImageServlet.java]: Make folder configurable with for example system property

File: src/main/java/fi/metatavu/metamind/images/ImageServlet.java [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+      if ( knot == null ) {
+        resp.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+        return;
+      }
+      
+      String extension = file.getSubmittedFileName().substring(file.getSubmittedFileName().lastIndexOf("."));
+      String fileURL = "images/"+knotId.toString()+extension;
+      InputStream fileInputStream = file.getInputStream();
+      
+      File dir = new File("images");
```
Comment (hkurhinen) [inline src/main/java/fi/metatavu/metamind/images/ImageServlet.java]: Make folder configurable with for example system property

File: src/main/java/fi/metatavu/metamind/persistence/dao/KnotDAO.java [hunk 4]
```diff
@@ -150,5 +150,9 @@ public Knot updateStory(Knot knot, Story story, UUID lastModifierId) {
     knot.setStory(story);
     return persist(knot);
   }
+
```
Comment (hkurhinen) [inline src/main/java/fi/metatavu/metamind/persistence/dao/KnotDAO.java]: Remove unnecessary empty lines

File: src/main/resources/META-INF/persistence.xml [hunk 5]
```diff
@@ -4,6 +4,7 @@
 
   <persistence-unit name="metamind" transaction-type="JTA">
     <jta-data-source>java:jboss/datasources/metamind-api</jta-data-source>
+    <class>fi.metatavu.metamind.images.ImagesServlet</class>
```
Comment (hkurhinen) [inline src/main/resources/META-INF/persistence.xml]: Is this necessary here?

**Review by hkurhinen — CHANGES_REQUESTED** [review]

File: src/main/resources/META-INF/persistence.xml (hunk 5 above)
Comment (platonovsimeon) [reply src/main/resources/META-INF/persistence.xml]: No

Comment (platonovsimeon) [reply src/main/resources/META-INF/persistence.xml]: Removed

File: src/main/java/fi/metatavu/metamind/persistence/dao/KnotDAO.java (hunk 4 above)
Comment (platonovsimeon) [reply src/main/java/fi/metatavu/metamind/persistence/dao/KnotDAO.java]: Removed

File: src/main/java/fi/metatavu/metamind/images/ImageServlet.java (hunk 1 above)
Comment (platonovsimeon) [reply src/main/java/fi/metatavu/metamind/images/ImageServlet.java]: Folder can now be configured using system property "metamind.imagedir"

File: src/main/java/fi/metatavu/metamind/images/ImageServlet.java (hunk 3 above)
Comment (platonovsimeon) [reply src/main/java/fi/metatavu/metamind/images/ImageServlet.java]: Folder can now be configured using system property "metamind.imagedir"

File: src/main/java/fi/metatavu/metamind/images/ImageServlet.java (hunk 2 above)
Comment (platonovsimeon) [reply src/main/java/fi/metatavu/metamind/images/ImageServlet.java]: Folder can now be configured using system property "metamind.imagedir"

**Pull request closed without merging**